llvm/test/Transforms/InstCombine/consecutive-fences.ll

```diff
@@ -30,6 +30,15 @@ define void @test_target_dependent_scope() {
   ret void
 }
 
+; CHECK-LABEL: combine_target_dependent_scope
+; CHECK-NEXT: fence syncscope("agent") seq_cst
+; CHECK-NEXT: ret void
+define void @combine_target_dependent_scope() {
+  fence syncscope("agent") acq_rel
+  fence syncscope("agent") seq_cst
+  ret void
+}
+
```

Comment:
Add tests for MMRA preservation (e.g. both same MMRA, and one with different, one missing)

Reply:
While it's safe to ignore and drop all MMRA, the ideal thing to do is compute the correct MMRA for the final fence. I don't know that is already happening in this path of InstCombine.

@Pierre-vh

Reply:
I don't think it is combined here, we'd need to call `MMRAMetadata::combine` or `combineMetadata` from Local.h